src/components/UriViewer.js

```diff
@@ -120,6 +121,11 @@ const DownloadButton = ({ uri, metadata: { bucket, name, fileName, size } }) =>
       div({ style: { display: 'flex', justifyContent: 'center' } }, [
         h(ButtonPrimary, {
           disabled: !url,
+          onClick: () => {
+            if (uri.toLowerCase().endsWith('.bam')) {
+              Ajax().Metrics.captureEvent(Events.workspaceDataDownload, { downloadFrom: 'bam file' })
+            }
+          },
```

Comment:
```suggestion
          onClick: uri.toLowerCase().endsWith('.bam') ?
            () => Ajax().Metrics.captureEvent(Events.workspaceDataDownload, { downloadFrom: 'bam file' }):
            undefined,
```

Reply:
For my own edification, what is the advantage of this change?

Reply:
This way, if it's not a bam file, the onClick will be undefined rather than a no-op, so it doesn't even have to call it to find out.

Reply:
Does that mean at render time we're calling endsWith() on all the url strings in the table?  If so, that seems like we might be trading off a slower render time for faster click handling.  I'm an infrequent front end developer, so if your instincts tell you this is right way to do it, I'll be quiet :)

Reply:
Ah, so because this is in the modal, and that's not rendered until it's mounted, it'll only run once per render within that, and it's cheap enough that I'd tell you to do it this way for consistency. A good thought, though!

Reply:
I see! I was conflating the other cases and forgot this was in the modal.

Reply:
I spoke with Janki and she would like us to capture downloads for the other file types so this conditional is obsolete now I think.

Reply:
Maybe something like this to capture the extension?
```js
_.tail(/\.\w+$/.exec(uri))
```